server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -97,7 +97,12 @@ public FsBlobContainer(FsBlobStore blobStore, BlobPath blobPath, Path path) {
         blobNamePrefix = blobNamePrefix == null ? "" : blobNamePrefix;
         try (DirectoryStream<Path> stream = Files.newDirectoryStream(path, blobNamePrefix + "*")) {
             for (Path file : stream) {
-                final BasicFileAttributes attrs = Files.readAttributes(file, BasicFileAttributes.class);
+                final BasicFileAttributes attrs;
+                try {
+                    attrs = Files.readAttributes(file, BasicFileAttributes.class);
+                } catch (FileNotFoundException e) {
+                    continue;
+                }
                 if (attrs.isRegularFile()) {
```

Comment:
perhaps easier as the two suggestions above is to use `Files.isRegularFile(file)` here which takes care of all this.

Reply:
This won't help us I think? We still want the file size two lines below where we'd have to make another call to the FS that we'd have to guard against "file not found".